layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayout.kt

```diff
@@ -82,35 +86,30 @@ open class FoldableLayout @JvmOverloads constructor(
             createView(config)
         }
 
-        setupViewModel()
         registerWindowInfoFlow()
     }
 
-    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
-        super.onSizeChanged(w, h, oldw, oldh)
-        this.postDelayed(
-            {
-                addContent()
-                requestLayout()
-            },
-            50L
-        )
-    }
-
     private fun registerWindowInfoFlow() {
-        val windowInfoRepository: WindowInfoRepository =
-            (context as Activity).windowInfoRepository()
-        job = MainScope().launch {
-            windowInfoRepository.windowLayoutInfo
-                .collect { info ->
+        val activity = context as AppCompatActivity
+        val windowInfoRepository = activity.windowInfoRepository()
+        activity.lifecycleScope.launch(Dispatchers.Main) {
+            activity.lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
```

Comment:
`RESUMED` or `STARTED`?

Reply:
In our case It's working also with RESUMED or STARTED, but I changed it to STARTED.